libtiledbsoma/src/soma/soma_dimension.h

```diff
@@ -0,0 +1,147 @@
+/**
+ * @file   soma_dimension.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ *   This file defines the SOMADimension class. SOMADimension acts as a wrapper
+ * to a TileDB Dimension and implements function to perform queries as well as
+ * core domain and current domain operations. It provides a common interface
+ * identical to TileDB attributes and composite columns.
+ */
+
+#ifndef SOMA_DIMENSION_H
+#define SOMA_DIMENSION_H
+
+#include <algorithm>
+#include <vector>
+
+#include <tiledb/tiledb>
+#include "soma_column.h"
+
+namespace tiledbsoma {
+
+using namespace tiledb;
+
+class SOMADimension : public SOMAColumn {
+   public:
+    static std::shared_ptr<SOMADimension> create(
+        std::shared_ptr<Context> ctx,
+        ArrowSchema* schema,
+        ArrowArray* array,
+        const std::string& soma_type,
+        std::string_view type_metadata,
+        PlatformConfig platform_config);
+
+    SOMADimension(Dimension dimension)
+        : dimension(dimension) {
+    }
+
+    virtual inline std::string name() const override {
```

Comment:
You should drop the `virtual` from these methods. See [ISO C++ Core Guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final) for an explanation.